packages/firestore/src/core/bundle.ts

```diff
@@ -265,3 +257,113 @@ export class BundleLoader {
     return new BundleLoadResult({ ...this.progress }, changedDocuments);
   }
 }
+
+export async function firestoreClientLoadBundle(
+  client: FirestoreClient,
+  data: ReadableStream<Uint8Array> | ArrayBuffer | string,
+  resultTask: LoadBundleTask
+): Promise<void> {
+  const reader = createBundleReader(
+    data,
+    newSerializer((await client.getConfiguration()).databaseInfo.databaseId)
+  );
+  client.asyncQueue.enqueueAndForget(async () => {
+    syncEngineLoadBundle(await getSyncEngine(client), reader, resultTask);
+  });
+}
+
+export function firestoreClientGetNamedQuery(
```

Comment:
These two methods should live in `firestore_client` along with all their counterparts. In Android and iOS, they will be part of the FIrestoreClient class.

Reply:
Done.

I leave the refactor for local_store and sync_engine alone, because they actually look good.